apps/web/src/services/asset-inventory/collector/shared/collector-data-modal/collector-data-modal-store.ts

```diff
@@ -1,24 +1,43 @@
 import { defineStore } from 'pinia';
 
-import type { CollectorModel, SecretModel, JobStatus } from '@/services/asset-inventory/collector/model';
+import { SpaceConnector } from '@cloudforet/core-lib/space-connector';
+
+import ErrorHandler from '@/common/composables/error/errorHandler';
+
+import type {
+    CollectorModel, SecretModel, JobModel,
+} from '@/services/asset-inventory/collector/model';
 import type { CollectDataType } from '@/services/asset-inventory/collector/shared/collector-data-modal/type';
 import { COLLECT_DATA_TYPE } from '@/services/asset-inventory/collector/shared/collector-data-modal/type';
 
-interface RecentJob {
-    jobId: string;
-    status: JobStatus;
-    secretId?: string;
-}
-
 export const useCollectorDataModalStore = defineStore('collector-data-modal', {
     state: () => ({
+        recentJob: undefined as JobModel|null|undefined,
         // Required
         visible: false, // Determine the visibility of the collector-data-modal.
-        recentJob: undefined as RecentJob|null|undefined, // Determine the theme-color of the p-button-modal, duration, status.
         selectedCollector: undefined as CollectorModel|null|undefined, // This state is used for API.
         collectDataType: COLLECT_DATA_TYPE.ENTIRE as CollectDataType, // This is the state for distinguishing between overall collection and collection per account.
 
         // Optional
         selectedSecret: undefined as SecretModel|undefined, // In detail page, This state is used for API.
     }),
+    actions: {
+        async getJobs(collectorId: string) {
+            try {
+                const res = await SpaceConnector.client.inventory.job.list({ collector_id: collectorId });
+                if (res.results.length > 0) {
+                    if (this.selectedSecret) {
+                        const filteredJobs = res.results.filter((job) => job.secret_id);
+                        this.recentJob = filteredJobs[0];
+                    } else {
+                        const filteredJobs = res.results.filter((job) => !job.secret_id);
+                        this.recentJob = filteredJobs[0];
+                    }
+                }
+            } catch (e) {
+                ErrorHandler.handleError(e);
+                throw e;
```

Comment:
question: I wonder why it throws the error after handling the error. 😮

Reply:
removed it! thx!